### ISSUE TITLE
[Bug] Wrong behaviour leaving the address bar

### ISSUE THREAD

**Issue by theFireProjet** [mozilla-mobile/fenix#9445]
## Steps to reproduce
The bug only occurs occasionally which makes it hard to reproduce. I had to try multiple times until it happened:
- tap the address bar while on a website (in my case: https://github.com/mozilla-mobile/fenix/issues),
- very quickly tap 'Back' (only once) as address bar and keyboard are opening.

At some occasions the bug seems to occur more often, while it does only rarely at others.
Maybe this has to do with the complexity of the current website, but that's only an assumption:
- The bug rarely occured viewing a Wikipedia article.
- It occured more often on: https://github.com/mozilla-mobile/fenix/issues.
- It feels like it occurs most often when writing a new issue looking at this website with the text field:
https://github.com/mozilla-mobile/fenix/issues/new?assignees=&labels=%F0%9F%90%9E+bug&template=---bug-report.md&title=%5BBug%5D.

### Expected behavior
- If the 'Back' button is facing left, that means keyboard is not yet opened and the address bar is displayed together with the search engines:
Tapping 'Back' one time is supposed to close the search bar/address bar view. Upcoming keyboard should not appear or instantly disappear again due to now more editing the address bar.
- If the 'Back' button is facing downwards, meaning keyboard is opened or about to open and address bar is displayed together with search engines:
Tapping 'Back' one time is supposed to stop the keyboard from showing up or close it again. In result Firefox is supposed to leave the search view as it usually does with the keyboard being closed.

### Actual behavior
There are three different variants of wrong behaviour after pressing 'Back':
- Firefox goes back to the website but the keyboard still opens/stays open:
1st video file (20200327_001508 1.mp4)
<video src="https://srv-file5.gofile.io/download/TOo5Us/20200327_001508%201.mp4" width="320" height="200" controls preload></video>
- Firefox goes to the tabs overview:
2nd video file (20200327_003106 1.mp4)
<video src="https://srv-file5.gofile.io/download/TOo5Us/20200327_003106%201.mp4" width="320" height="200" controls preload></video>
- Firefox goes back to the website view but in addition it goes one site back. This occurs even when going back from the address bar at any time, not very quickly:
3rd video file (20200327_003320 1.mp4)
<video src="https://srv-file5.gofile.io/download/TOo5Us/20200327_003320%201.mp4" width="320" height="200" controls preload></video>

Screen recordings: https://gofile.io/?c=TOo5Us (outdated link)

### Device information

* Android device: Samsung Galaxy S9+ (Android 8.0.0)
* Fenix version: 4.1.0

**Comment by ekager** [comment]
@theFireProjet are you still seeing this issue?

**Comment by theFireProjet** [comment]
> 
> 
> @theFireProjet are you still seeing this issue?

I cannot reproduce any issue like that at the moment. Possibly the behaviour when interacting with the address bar has changed anyways since back then. Now it looks like there is no problem anymore.
